scripts/checker_commons.py

```diff
@@ -1,8 +1,64 @@
-import json, sys
+import json, os, sys
 from collections import defaultdict
+from multiprocessing import cpu_count, Pool
 
+try:  # optional dependency to display a progress bar
+    from tqdm import tqdm
 
-def aggregate(pdf_filepath, report, aggregated_report_filepath):
+    HIDE_STDERR = True
+except ImportError:
+    tqdm = lambda _, total: _
+    HIDE_STDERR = False
+
+
+def main(checker_name, analyze_pdf_file, argv, checks_details_url):
+    if len(argv) != 2:
+        print(argv, file=sys.stderr)
+        print(
+            f"Exactly one argument must be passed to {checker_name}.py", file=sys.stderr
+        )
+        sys.exit(2)
+    elif argv[1] == "--print-aggregated-report":
+        print_aggregated_report(checker_name, checks_details_url)
+    elif argv[1] == "--process-all-test-pdf-files":
+        process_all_test_pdf_files(checker_name, analyze_pdf_file)
+    else:
+        print(analyze_pdf_file(argv[1]))
+
+
+def process_all_test_pdf_files(checker_name, analyze_pdf_file):
+    pdf_filepaths = [
```

Comment:
Nitpick: I think this can be archived with less code using `pathlib.Path.glob()` https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

Reply:
Thank you for your feedback @Tolker-KU 🙂

`scantree()` uses `os.scandir()` that is very efficient for this task: https://docs.python.org/3/library/os.html#os.scandir

> The [scandir()](https://docs.python.org/3/library/os.html#os.scandir) function returns directory entries along with file attribute information, giving better performance for many common use cases.

Do you think that `glob()` would be as efficient?

Reply:
`pathlib` will probably have a small amount of overhead from construction of the `Path` objects, but it uses `os.scandir()` under the hood for walking the file system:

https://github.com/python/cpython/blob/b0202a4e5d6b629ba5acbc703e950f08ebaf07df/Lib/pathlib.py#L1042-L1047
https://github.com/python/cpython/blob/b0202a4e5d6b629ba5acbc703e950f08ebaf07df/Lib/pathlib.py#L1057-L1122
https://github.com/python/cpython/blob/b0202a4e5d6b629ba5acbc703e950f08ebaf07df/Lib/pathlib.py#L166-L191
https://github.com/python/cpython/blob/b0202a4e5d6b629ba5acbc703e950f08ebaf07df/Lib/pathlib.py#L1013-L1017

Reply:
OK, nice to know!

I think I'll stick with this for this PR,
but i'll be happy to merge another PR that improve/shorten this code 🙂